libs/sdk-ui-dashboard/src/model/commandHandlers/alerts/updateAlertHandler.ts

```diff
@@ -0,0 +1,43 @@
+// (C) 2021 GoodData Corporation
+import { SagaIterator } from "redux-saga";
+import { call, put } from "redux-saga/effects";
+import { IWidgetAlert } from "@gooddata/sdk-backend-spi";
+import { objRefToString } from "@gooddata/sdk-model";
+import { dispatchDashboardEvent } from "../../eventEmitter/eventDispatcher";
+import { DashboardContext } from "../../types/commonTypes";
+import { UpdateAlert } from "../../commands/alerts";
+import { alertUpdateSuccess, alertUpdateError } from "../../events/alerts";
+import { PromiseFnReturnType } from "../../types/sagas";
+import { alertsActions } from "../../state/alerts";
+
+function updateAlert(ctx: DashboardContext, alert: IWidgetAlert): Promise<IWidgetAlert> {
+    const { backend, workspace } = ctx;
+
+    return backend.workspace(workspace).dashboards().updateWidgetAlert(alert);
+}
+
+export function* updateDashboardAlertCommandHandler(
+    ctx: DashboardContext,
+    cmd: UpdateAlert,
+): SagaIterator<void> {
+    // eslint-disable-next-line no-console
+    console.debug("handling update alert", cmd, "in context", ctx);
+
+    try {
+        const alert: PromiseFnReturnType<typeof updateAlert> = yield call(
+            updateAlert,
+            ctx,
+            cmd.payload.alert,
+        );
+
+        yield put(
+            alertsActions.updateAlert({
+                changes: cmd.payload.alert,
+                id: objRefToString(cmd.payload.alert.ref),
+            }),
+        );
+        yield dispatchDashboardEvent(alertUpdateSuccess(ctx, alert, cmd.correlationId));
+    } catch (e) {
+        yield dispatchDashboardEvent(alertUpdateError(ctx, cmd.payload.alert, e, cmd.correlationId));
```

Comment:
same here, please use the more 'generic' command failed event with internal_error reason